### PR TITLE
feat:Update api to include role category (RWA-944)

### PR DESCRIPTION
**Pull request by dharmendrak**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-944

### Change description ###

Update api to include role category

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by serdaroguzhanakin — DISMISSED** [review]

**Review by indvamshi — DISMISSED** [review]

**Review by BenNzewi — COMMENTED** [review]
This needs to be rebased with 957 so we can see the actual work done so far.

**Review by BenNzewi — CHANGES_REQUESTED** [review]
I cross referenced RWA-957 and the changes there for for including description on all our search task endpoints look incorrect. It should be mapped to the task and surfaceable without having to hard code.

The FTs will need reviewing

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/TaskManagementGetTaskBySearchForCompletablePactTest.java [hunk 1]
```diff
@@ -164,7 +165,7 @@ public void getTasksBySearchForCompletableCriteriaWithWarnings() {
                         PermissionTypes.MANAGE,
                         PermissionTypes.REFER
                     ))),
-            "LEGAL_OPERATIONS",
+            RoleCategory.LEGAL_OPERATIONS.name(),
```
Comment (BenNzewi) [inline src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/TaskManagementGetTaskBySearchForCompletablePactTest.java:168]: can we also update the local consumer to include role_category? TaskManagementGetTasksBySearchForCompletableConsumerTest

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/TaskManagementGetTaskProviderTest.java [hunk 2]
```diff
@@ -160,7 +161,7 @@ private Task createTask() {
                         PermissionTypes.MANAGE,
                         PermissionTypes.REFER
                     ))),
-            "LEGAL_OPERATIONS",
+            RoleCategory.LEGAL_OPERATIONS.name(),
```
Comment (BenNzewi) [inline src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/TaskManagementGetTaskProviderTest.java:164]: can we also update the local consumer to include role_category?

TaskManagerGetTaskConsumerTest

**Review by BenNzewi — COMMENTED** [review]
is there a ticket to update the post-deployment-ft scenarios with role category?

**Review by BenNzewi — APPROVED** [review]

**Pull request merged**